src/sha512.rs

```diff
@@ -123,6 +132,13 @@ impl HashTrait for Hash {
         Hash(e.midstate())
     }
 
+    #[cfg(feature = "fuzztarget")]
+    fn from_engine(e: HashEngine) -> Hash {
+        let mut hash = e.midstate();
+        hash[0] ^= 0xff; // Make this distinct from SHA-256
```

Comment:
A little confused why this is needed... Seems like it's modified, not used, then dropped?

Reply:
Well it would help if I constructed a valid Hash(), but this is to make it consistent with the current rust-bitcoin SHA-512, which inits with 0xff.